src/pages/amazon-sales-channel/best-practices/database-storage.md

```diff
@@ -0,0 +1,129 @@
+  "orange": ["id-5", "id-6"]
+}
+```
+
+You can improve on this further by creating a key/value pair for each type. This results with faster indexing. Retrieve the IDs by type and quickly access the key/value storage to obtain the entities. The downside of this approach is the need the maintain an index.
+
+**No ability to select or query columns or put limit on results returned**.
+
+This can be avoided by maintaining an index of keys. Example would be a simple array of IDs related to the entity:
+
+`const entities = ["id-1", "id-2", "id-3"];`
+
+Using an index of keys enables you to create utility functions that would divide the list of entities into separate chunks. This practice creates a limit on the number of returned items and uses the page as a cursor to get the next set of results when queried. For example:
+
+```js
+export function findPage<T>(
+  collection: Array<T>,
+  total: number,
+  currentPage: number,
+  itemsPerPage: number,
+): Array<T> {
+  const fromSlice = (currentPage - 1) * itemsPerPage;
+  let toSlice = fromSlice + itemsPerPage;
+
+  if (toSlice > total) {
+ | toSlice = total;
+  }
+
+  return collection.slice(fromSlice, toSlice);
+}
+
+export async function pageable<T>(
+  ids: Array<ID>,
+  repository: PageableRepository<T>,
+  options: PageableOptions,
+  page?: number,
+): Promise<Collection<T>> {
+  const { defaultPage, itemsPerPage } = options;
+
+  const total = ids.length;
+  const currentPage = page ?? defaultPage;
+
+  const subsetIds = findPage<string>(ids, total, currentPage, itemsPerPage);
+
+  const promises: Array<Promise<T>> = [];
+  subsetIds.forEach((id: ID) => {
+    promises.push(repository.getEntity(id));
+  });
+
+  const collection: Array<T> = await Promise.all(promises);
+
+  return {
+    total,
+    collection,
+    count: collection.length,
+    pagination: {
+      currentPage,
+      lastPage: Math.ceil(total / itemsPerPage),
+    },
+  };
+}
+```
+
+## Adobe I/O File library
+
+There are benefits and limitations in using the Adobe I/O File library.
+
+### Benefits
+
+Since `lib-files` uses blob storage in the Azure cloud, each user has their own segregated buckets. This is more segregated than `lib-state`, which is a shared Cosmos DB cluster.
+
+The development team also found that while `lib-state` is great at read/write operations, `lib-files` is not. The file library is a great option for tasks that are not read/write heavy. Since credentials require the utmost security, it did not seem prudent to use `lib-state` for this use case as it's not multi-tenant.Therefore, reading credentials from lib-files was a better option overall.
+
+### Limitations
+
+The `lib-files` library has some limitations from both practice and the public documentation. Its closest comparison is Amazon S3 storage.
+
+* Read/write operations are not suitable for a high throughput layer. If you need to run a large number of operations, it can take roughly ten to 20 seconds to return sizable results.
```

Comment:
```suggestion
* Read/write operations are not suitable for a high throughput layer. If you need to run a large number of operations, it can take roughly 10 to 20 seconds to return sizable results.
```